src/us/kbase/workspace/test/database/provenance/CreditMetadataEntryTest.java

```diff
@@ -0,0 +1,256 @@
+			Title.getBuilder("\t\t\f\t\n\rA Series of Unfortunate Elephants\n\n\n   ")
+					.build());
+
+	private static final CreditMetadata CREDIT_METADATA_EXAMPLE = CreditMetadata
+			.getBuilder("some:boring_identifier",
+					ResourceType.DATASET,
+					CONTRIBUTOR_LIST,
+					TITLE_LIST)
+			.withVersion("1.0.1-fml")
+			.build();
+
+	private static final String[] INVALID_SEM_VER_STRINGS = {
+			"v1",
+			"foo",
+			"01.1.0",
+			"alpha.01",
+			"beta1..1",
+			"build.01",
+			".build.01" };
+
+	private static final String EXP_EXC = "expected exception";
+
+	@Test
+	public void equals() throws Exception {
+		EqualsVerifier.forClass(CreditMetadataEntry.class).usingGetClass().verify();
+	}
+
+	/**
+	 * Check the resource metadata has the expected fields
+	 *
+	 * @param CreditMetadataEntry
+	 *                instance to check
+	 * @param user
+	 *                workspace user
+	 * @param version
+	 *                creditMetadataSchemaVersion string
+	 * @param creditMetadata
+	 *                instance to check
+	 */
+	private void assertResourceMetadataFields(
+			final CreditMetadataEntry creditMetadataEntry,
+			final WorkspaceUser user,
+			final String version,
+			final CreditMetadata creditMetadata) {
+		assertThat(INCORRECT_USER, creditMetadataEntry.getUser(), is(user));
+		assertThat(INCORRECT_SCHEMA_VERSION,
+				creditMetadataEntry.getCreditMetadataSchemaVersion(),
+				is(version));
+		TestCommon.assertCloseToNow(creditMetadataEntry.getTimestamp());
+		assertThat(INCORRECT_CREDIT_METADATA,
+				creditMetadataEntry.getCreditMetadata(),
+				is(creditMetadata));
+	}
+
+	@Test
+	public void buildResourceMetadata() throws Exception {
+		final WorkspaceUser user = new WorkspaceUser("worst_ws_user");
+		final CreditMetadataEntry cmc = CreditMetadataEntry.build(
+				user,
+				VERSION_STRING_UNTRIMMED,
+				Instant.now(),
+				CREDIT_METADATA_EXAMPLE);
+		assertResourceMetadataFields(cmc, user, VERSION_STRING, CREDIT_METADATA_EXAMPLE);
+	}
+
+	/**
+	 * Checks the error thrown when a ResourceMetadata object is built with invalid
+	 * arguments
+	 *
+	 * @param builder
+	 *                resource metadata builder, complete with args
+	 * @param error
+	 *                the expected error
+	 */
+	private void buildResourceMetadataFailWithError(
+			final WorkspaceUser wsUser,
+			final String version,
+			final CreditMetadata creditMetadata,
+			final Instant timestamp,
+			final String errorString) {
+		try {
+			CreditMetadataEntry.build(wsUser, version, timestamp, creditMetadata);
+			fail(EXP_EXC);
+		} catch (Exception got) {
+			TestCommon.assertExceptionCorrect(got, new IllegalArgumentException(
+					"Errors in CreditMetadataEntry construction:\n"
+							+ errorString));
+		}
+	}
+
+	@Test
+	public void buildFailNullUser() throws Exception {
+		buildResourceMetadataFailWithError(
+				null,
+				VERSION_STRING_UNTRIMMED,
+				CREDIT_METADATA_EXAMPLE,
+				Instant.now(),
+				USER_NON_NULL);
+		buildResourceMetadataFailWithError(
```

Comment:
This test is redundant since there's no ambiguity for the type of the user argument

Reply:
Same for other tests below

Reply:
:+1: